sentinelhub/config.py

```diff
@@ -191,71 +190,56 @@ def load(cls, filename: str) -> SHConfig:
         return config
 
     def save(self, filename: Optional[str] = None) -> None:
-        """Method that saves configuration parameter changes from instance of SHConfig class to global config class and
-        to `config.json` file.
+        """Saves configuration parameters to the user settings in the `config.json` file.  If a filename is not
+        specified the configuration is saved to the location specified by `SHConfig.get_config_location()`.
 
-        :param filename: Optional name of file to which to save configuration. If not specified saves to global default.
-
-        :Example:
-            ``my_config = SHConfig()`` \n
-            ``my_config.instance_id = '<new instance id>'`` \n
-            ``my_config.save()``
+        :param filename: Optional name of file to which to save configuration.
         """
         self._validate_values()
 
-        is_changed = False
-        for param in self.CONFIG_PARAMS:
-            if getattr(self, param) != self._global_cache[param]:
-                is_changed = True
-                self._global_cache[param] = getattr(self, param)  # pylint: disable=unsupported-assignment-operation
+        config_dict = {param: getattr(self, param) for param in self.CONFIG_PARAMS}
```

Comment:
probably, but also likely a case of parallel development.
do you think it would be worthwhile to have a `_to_dict` method for this? or is the onliner simple enough that we don't care that it's used in two places?

Reply:
It's simple enough IMHO